### PR TITLE
Cheat: Allow placing track at any height

### PR DESCRIPTION
**Pull request by bitman2049**
Add a cheat to enable building track at all height intervals.

Issue link: https://github.com/OpenRCT2/OpenRCT2/issues/10572

### CODE REVIEW

**Comment by janisozaur** [comment]
Thanks for your PR. We're currently working on getting as many fixes as we can to do another release and it will be a while longer until we will consider merging your feature.

**Comment by duncanspumpkin** [comment]
Why have this as a separate option? In what situation does someone want this but not full support clearance off

**Comment by deurklink** [comment]
@duncanspumpkin Clearance checks are usually disabled on our server. We often make structures by putting many monorail walls on top of each other, for example. For that, this would be a huge timesaver. However, when building an actual roller coaster, we wouldn't want this option turned on, because it would be really annoying to find out later that you built your rollercoaster at a height where you would not be able to connect a path to it.

**Comment by duncanspumpkin** [comment]
@deurklink okay if you think it is necessary to have this as a separate option I'm not going to argue.

**Comment by Umdlye** [comment]
The cheat will be called "Allow building at any height" in the menu but you refer to it as building track at invalid heights, which is more descriptive IMO.

**Comment by tupaschoal** [comment]
This needs rebasing, I'd also take @Umdlye 's comment into consideration

**Comment by bitman2049** [comment]
I just rebased with latest

**Comment by Gymnasiast** [comment]
Needs changelog entry:
`Feature: [##10572] Cheat to allow building at invalid heights.`

Otherwise: LGTM.

File: src/openrct2-ui/windows/Cheats.cpp [hunk 1]
```diff
@@ -1074,6 +1077,9 @@ static void window_cheats_rides_mouseup(rct_window* w, rct_widgetindex widgetInd
         case WIDX_ENABLE_ALL_DRAWABLE_TRACK_PIECES:
             CheatsSet(CheatType::EnableAllDrawableTrackPieces, !gCheatsEnableAllDrawableTrackPieces);
             break;
+        case WIDX_ALLOW_TRACK_PLACE_INVALID_HEIGHTS:
+            CheatsSet(CheatType::AllowTrackPlaceInvalidHeights, !gCheatsAllowTrackPlaceInvalidHeights);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Cheats.cpp:1081]: Put this above:
```
            if (!gCheatsAllowTrackPlaceInvalidHeights)
            {
                context_show_error(STR_WARNING_IN_CAPS, STR_THIS_FEATURE_IS_CURRENTLY_UNSTABLE);
            }
```

**Comment by bitman2049** [comment]
K, made those changes and resolved any conflicts. Should be good to go

File: distribution/changelog.txt [hunk 2]
```diff
@@ -14,6 +14,7 @@
 - Fix: RCT1 scenarios have more items in the object list than are present in the park or the research list.
 - Improved: [#6530] Allow water and land height changes on park borders.
 - Improved: [#11390] Build hash written to screenshot metadata.
+- Feature: [#10572] Cheat to allow building at invalid heights.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:17]: Should be grouped together with the other features.

Comment (bitman2049) [reply distribution/changelog.txt]: Got it, fixed now.

File: distribution/changelog.txt [hunk 3]
```diff
@@ -3,6 +3,7 @@
 - Feature: [#9029] Open doors with the tile inspector.
 - Feature: [#11231] Change shortcut window list order to be more intuitive, and split it into logical sections.
 - Feature: [#11306] Path additions are now kept when replacing the path.
+- Feature: [#10572] Cheat to allow building at invalid heights.
```
Comment (AaronVanGeffen) [inline distribution/changelog.txt:6]: Keep feature ids in sequential order:

```suggestion
- Feature: [#10572] Cheat to allow building at invalid heights.
- Feature: [#11231] Change shortcut window list order to be more intuitive, and split it into logical sections.
- Feature: [#11306] Path additions are now kept when replacing the path.
```

**Pull request merged**